adapter/PreciceInterface.c

```diff
@@ -700,6 +700,8 @@ void PreciceInterface_ConfigureCouplingData(PreciceInterface *interface, Simulat
   interface->numWriteData = config->numWriteData;
   if (config->numWriteData > 0)
     interface->writeData = malloc(config->numWriteData * sizeof(int));
+  else
+    interface->writeData = NULL;
```

Comment:
I am not familiar with the codebase or sure about good practices in C, but I understand that will make sure that the free at line 769 is not triggered when no write-data is configured. I guess one could default this to `NULL` in the structure definition.

Shouldn't the same also apply to `readData`?

Applying this change does not fix the issue for me, neither in debug nor in release mode. Does this actually work for you? Does it work on the Demo VM or in some CI environment?

Reply:
To solve the error, I needed this PR and some changes to the tutorial case that I just pushed (I had forgotten yesterday): CalculiX time & preCICE end time were longer than OpenFOAM end time.
Does it work for you with this PR + updated tutorial?

(Might be interesting to generate better errors when time don't match, still).

I moved the change to initialization and also initialized `readData`.